esmvalcore/_recipe/check.py

```diff
@@ -238,7 +239,7 @@ def check_for_temporal_preprocs(profile):
 def extract_shape(settings):
     """Check that `extract_shape` arguments are valid."""
     shapefile = settings.get('shapefile', '')
-    if not os.path.exists(shapefile):
+    if not Path(shapefile).exists():
```

Comment:
Note that Path(x) is a lot more fussy about what you put in than os.path.exists, so users may already get an exception from the error in creating a Path object from what they have defined instead of the intended RecipeError